### DISCUSSION TITLE
Recommended hardware

### DISCUSSION THREAD

**Discussion by ant-thomas** [open3e/open3e#150] [General]
I'm looking to use Open3E to interface with and control some elements of my Vitocal 150-A ASHP to avoid using the ViCare API. 

I've got a spare RPi I can use - is there a recommended CAN adapter to use?
Plan is to buy a USB adapter. And before I find and buy the cheapest one is there anything specific to look out for?

Also, does anyone know where to buy a suitable 6 pin connector to plug into the CAN connector on the unit?
(I'm in the UK)

I've seen posts mentioning interfacing using AP mode - is it possible to interface directly over my local network and avoid the CAN adapter/connection?

**Comment by MyHomeMyData** [comment]
> I've got a spare RPi I can use - is there a recommended CAN adapter to use?

I'm using [this one](https://amzn.eu/d/8Qx7TeT) and can recommend it. An important feature is voltage isolation and 2500V ESD isolated protection on CAN bus pins.

> I've seen posts mentioning interfacing using AP mode - is it possible to interface directly over my local network and avoid the CAN adapter/connection?

You have to bring your Viessmann device in AP mode and connect your computer (laptop, RPI, ...) to this AP. Dependent on E3 device it's not possible to have an internet connection in parallel. So that's nice to give it a try, but from my point of view it's not feasable for normal day-by-day use. CAN bus is much better.

**Comment by AmansRevenger** [comment]
Hi

if you check the (albeit german) Wiki you will find some answers

https://github.com/open3e/open3e/wiki/020-Inbetriebnahme-CAN-Adapter-am-Raspberry

[this](https://www.amazon.de/dp/B07Q812QK8?ref=ppx_yo2ov_dt_b_product_details&th=1) is the amazon link to the adapter I am currently using

For the CAN connector itself, you need [this](https://www.3-liter-heizung.de/viessmann-steckersatz-7866718.html )one (the number 91)

I can attach some pictures of it

![image](https://github.com/user-attachments/assets/16cf5109-ab41-497f-a890-79b9dca78f88)
![image](https://github.com/user-attachments/assets/4290ebc5-09cf-4821-bf2f-f573a2a6aca8)
![image](https://github.com/user-attachments/assets/d5190958-120f-462b-b38d-a7be9a857b6c)

Then you connect the CANBAN/USB Adapter like this

![image](https://github.com/user-attachments/assets/882e6d0e-3bb3-4875-b3ea-8b2df4006236)
![image](https://github.com/user-attachments/assets/27a05856-bfc6-4f4d-9209-5c034a444054)
![image](https://github.com/user-attachments/assets/d5e184dc-fffc-4b32-9e42-c0ee3539a7d0)
 

Hope this helps!

**Comment by ant-thomas** [comment]
Thanks both for the replies.

I believe I need the left 6 pin adapter like the Vitocal 250-A
It may be that my installer has one they can send me as it wasn't needed for my installation.

![image](https://github.com/user-attachments/assets/f799bf00-8f3d-418f-9769-edad866844ca)

**Comment by ant-thomas** [comment]
Not managed to get hold of the proper connector, so have instead created my own pin connectors using some ferrules.

Connected with my USB CAN adapter and it seems to be working. Did the scan. Read a few parameters, changed hot water temperature.

Next step is to use a spare RPi for this and get it all communicating with my Home Assistant setup.

**Reply by philippoo66** [reply]
hm I'm not sure but looks as if some like this might fit https://www.pollin.de/p/leiterplattenanschlussklemme-5pol-rm5-08-mm-10a-452656 search for "CTB9209-5 "

**Comment by ant-thomas** [comment]
Now setup with the RPi. 
Using one of these adapters - https://www.aliexpress.com/item/1005005963727624.html

Currently running `open3e_depictSystem` and it's finding everything

**Reply by Philip-Wiege** [reply]
Congratulations! If depict system finds multiple ECUs such as the main ECU 0x680 of your Vitocal 150 you're good to go. If I remember correctly you are using the external CAN bus so you can read all data but writing to some special DIDs is only available in the internal bus between the indoor and outdoor unit.

**Reply by ant-thomas** [reply]
I can write/change parameters through the interface I'm using, which is great. Tested with hot water target temperature and central heating temperature.

I've also now got it feeding the data through to Home Assistant thanks to the work @Wolfgang-03 has done. Just need to do some English translations!

**Reply by Philip-Wiege** [reply]
Is there anything specific you need translated? I think the code and datapoints / DIDs are already in English. The setup guide however is mainly in German.
